### PR TITLE
Average, Median, Typical, Weighted indicator selectors

### PR DESCRIPTION
**Pull request by StefanoRaggi**

### CODE REVIEW

**Comment by mchandschuh** [comment]
Cool stuff @StefanoRaggi - where are the definitions for these? Are they arbitrarily named or are these common names?

**Comment by StefanoRaggi** [comment]
@mchandschuh These are 4 TA-Lib indicators, that were converted as selectors: http://ta-lib.org/function.html
Their original names are AVGPRICE, MEDPRICE, TYPPRICE, WCLPRICE.
NinjaTrader has the last three as well (named Median, Typical and Weighted).

File: Common/Field.cs [hunk 1]
```diff
@@ -57,6 +57,38 @@ public static partial class Field
...
+        /// </summary>
+        public static Func<BaseData, decimal> Typical
+        {
+            get { return TradeBarPropertyOrValue(x => (x.High + x.Low + x.Close) / 3m); }
+        }
+
+        /// <summary>
+        /// Defines an average price that is equal to (H + L + 2*C) / 4
+        /// </summary>
+        public static Func<BaseData, decimal> Weighted
```
Comment (mchandschuh) [inline Common/Field.cs]: Thoughts on naming this `WeightedClose` to closer match ninja?

Comment (StefanoRaggi) [inline Common/Field.cs]: Ninja calls it Weighted, TA-lib WeightedClose, you choose :)

Comment (mchandschuh) [inline Common/Field.cs]: Sounds good to me! Great stuff. I'm going to fix the travis build, looks like a test is failing and then merge this in

**Comment by mchandschuh** [comment]
Merged in https://github.com/QuantConnect/Lean/commit/3e10a7d4819badfb29f853d28752b83f2952f420

**Pull request closed without merging**